### PR TITLE
google-rtc-audio: rtnr: mt8195: cherry-pick up to the latest released version

### PR DESCRIPTION
**Pull request by MingJenTai**
This PR merges back the following commits from origin/main

4789d623a93e3c44e1bc6f808c7ef4b1cf5cd6fa Add RTNR to MT8195 topology 
05bd87da37092e8db0d741b6595a6eb9e7a29ffb Add RTNR library path for MT8195 
0d5180d804fdc42be0cc1aa4507d932d64c4ed46 Add MT8195 override config 
c8b375638c759b19969f8777bece3c1b159d6097 topology1: mt8195: support Google rtc audio process with RTNR
2e6bfbec1ccc069129b3f91212fc8e6af04284bb RTNR: Decouple switch-enable control from params blob 
2fd648efac7dbfb59d3e1300ae5de628812e2190 mux: Use a fixed size for mux_stream_data

### CODE REVIEW

**Comment by johnylin76** [comment]
@kuanhsuncheng this cherry-pick PR would need MTK's help to approve/merge.

**Comment by afq984** [comment]
Actually since the code works on main, if we find merging these commits too much hassle, we can branch off main to mt8195/v0.5 when we want to do a release build instead.
The branches are only used to give us a reference on what commit a blob was built with. Plus in the case of mt8195 being broken on main, we still get a base to work on for urgent fixes.

**Comment by afq984** [comment]
To clarify on my previous comment, by hassle I mean I don't know an easy way to compare (if there are any changes) between the commit upstream and the commit cherry-picked, at least on GitHub.

File: src/audio/rtnr/CMakeLists.txt [hunk 1]
```diff
@@ -2,7 +2,8 @@
 
 add_local_sources(sof rtnr.c)
 
-sof_add_static_library(SOF_RTK_MA_API rtklib/tgl/libSOF_RTK_MA_API.a)
-sof_add_static_library(Suite_rename rtklib/tgl/libSuite_rename.a)
-sof_add_static_library(Net rtklib/tgl/libNet.a)
-sof_add_static_library(Preset rtklib/tgl/libPreset.a)
+if(CONFIG_TIGERLAKE)
```
Comment (cujomalainey) [inline src/audio/rtnr/CMakeLists.txt:5]: this is going to be not pretty very quickly. Can you just put your libs in the root of the `third_party_libraries` folder in the root?

**Review by cujomalainey — APPROVED** [review]
just a comment for possible improvement on main

File: src/audio/rtnr/CMakeLists.txt (hunk 1 above)
Comment (MingJenTai) [reply src/audio/rtnr/CMakeLists.txt:5]: The libraries could be built by different core configs so I think still need place them under different folders for each platform.

**Pull request merged**